### PR TITLE
Airlock API - instrument the airlock with notifications events

### LINKED ISSUE #2125: Airlock - instrument the airlock with notifications events
As a TRE Developer
I want to instrument the code 
so that notifications events are sent

## Acceptance criteria

- [ ] TRE API send to the event grid notifications event
- [ ] Schema is agreed with the developer working on the notification mechanism (logic app)

### PR DESCRIPTION
**Pull request by anatbal**
# PR fixes #2125

## What is being addressed

Sending a new airlock notification even to a new airlock notification topic. This is done parallelly with developing the logic app that will subscribe to this event hub topic.

### CODE REVIEW

File: devops/scripts/setup_local_debugging.sh [hunk 1]
```diff
@@ -44,6 +46,15 @@ az eventgrid topic update \
   --public-network-access enabled \
   --inbound-ip-rules "${IPADDR}" allow
 
+
+echo "Adding local IP Address to ${EVENT_GRID_AIRLOCK_NOTIFICATION_TOPIC_NAME}."
```
Comment (eladiw) [inline devops/scripts/setup_local_debugging.sh:50]: nice addition

File: templates/core/terraform/api-webapp.tf [hunk 2]
```diff
@@ -27,31 +27,32 @@ resource "azurerm_app_service" "api" {
...
-    "RESOURCE_LOCATION"                          = azurerm_resource_group.core.location
-    "SWAGGER_UI_CLIENT_ID"                       = var.swagger_ui_client_id
-    "AAD_TENANT_ID"                              = "@Microsoft.KeyVault(SecretUri=${azurerm_key_vault_secret.auth_tenant_id.id})"
-    "API_CLIENT_ID"                              = "@Microsoft.KeyVault(SecretUri=${azurerm_key_vault_secret.api_client_id.id})"
-    "API_CLIENT_SECRET"                          = "@Microsoft.KeyVault(SecretUri=${azurerm_key_vault_secret.api_client_secret.id})"
-    "RESOURCE_GROUP_NAME"                        = azurerm_resource_group.core.name
-    "SUBSCRIPTION_ID"                            = data.azurerm_subscription.current.subscription_id
-    CORE_ADDRESS_SPACE                           = var.core_address_space
-    TRE_ADDRESS_SPACE                            = var.tre_address_space
+    "APPLICATIONINSIGHTS_CONNECTION_STRING"          = module.azure_monitor.app_insights_connection_string
```
Comment (eladiw) [inline templates/core/terraform/api-webapp.tf:30]: what was changed here. the diff is not clear

**Review by eladiw — APPROVED** [review]
lgtm

File: templates/core/terraform/api-webapp.tf (hunk 2 above)
Comment (anatbal) [reply templates/core/terraform/api-webapp.tf:30]: This is because of formatting, the addition is the notification queue

**Pull request merged**